js/ui/pivot_grid/ui.pivot_grid.area_item.js

```diff
@@ -402,23 +402,29 @@ export const AreaItem = Class.inherit({
     },
 
     setGroupWidth: function(value) {
-        if(value >= 0) {
-            this._groupWidth = value;
-            this._groupElement[0].style.width = value + 'px';
-        } else {
-            this._groupElement[0].style.width = value;
-        }
+        this.groupElement()[0].style.width = isNumeric(value) ? `${value}px` : value;
     },
 
     setGroupHeight: function(value) {
-        this._groupHeight = null;
+        this.groupElement()[0].style.height = isNumeric(value) ? `${value}px` : value;
+    },
 
-        if(value >= 0) {
-            this._groupHeight = value;
-            this._groupElement[0].style.height = value + 'px';
-        } else {
-            this._groupElement[0].style.height = value;
+    getGroupHeight: function() {
+        return this._getGroupElementSize('height');
+    },
+
+    getGroupWidth: function() {
+        return this._getGroupElementSize('width');
+    },
+
+    _getGroupElementSize(dimension) {
+        const size = this.groupElement()[0].style[dimension];
+
+        if(size.indexOf('px') >= 0) {
+            return Number(size.replace('px', ''));
```

Comment:
TODO: parseHeight/normalizeStyleProp/parsePixelValue ?

Reply:
none of these methods satisfy the current implementation